### PR TITLE
bump bom and jenkins version

### PR DESCRIPTION
**Pull request by jetersen**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by jetersen** [comment]
Could use a branch for backporting it to [277](https://github.com/jenkinsci/ssh-credentials-plugin/releases/tag/277.v95c2fec1c047)

@jglick

File: pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
 
   <properties>
     <changelist>999999-SNAPSHOT</changelist>
-    <jenkins.version>2.289.3</jenkins.version>
+    <jenkins.version>2.319.3</jenkins.version>
```
Comment (jglick) [inline pom.xml:74]: https://github.com/jenkinsci/ssh-credentials-plugin/pull/133 in the next release, https://github.com/jenkinsci/ssh-credentials-plugin/releases/tag/287.v0c2c2889c28d, bumps up to 2.346.x. So this update would be exposed to 2.319.x and 2.332.x users. OK I think, though it is unusual to change core dep in a backport branch.

**Comment by jglick** [comment]
`RequireUpperBoundDeps` error

**Comment by jglick** [comment]
https://github.com/jenkins-infra/jenkins-maven-cd-action/issues/4 :popcorn:

**Comment by jglick** [comment]
Going to have to wait until Monday I am afraid. :wave:

**Comment by jetersen** [comment]
Excuse me? Pug leaning back confused

> Skipping deployment as no artifacts were found with the expected path, typically due to a PR merge build not up to date with its base branch

@jglick no worries I can have an incremental... Hopefully

**Comment by jetersen** [comment]
Confirmed to work: https://github.com/jenkinsci/bom/pull/1275

**Comment by jglick** [comment]
https://github.com/jenkinsci/ssh-credentials-plugin/releases/tag/277.280.v1e86b_7d0056b_

**Pull request merged**